### PR TITLE
Adding persistent volume for tenants

### PR DESCRIPTION
**Pull request by mushthaq33**
## Purpose
Adding persistent volume for tenants

### CODE REVIEW

File: helm/pattern-1/apim-with-analytics/templates/wso2apim-volume-claim.yaml [hunk 1]
```diff
@@ -27,3 +27,21 @@ spec:
   selector:
     matchLabels:
       purpose: apim-shared-deployment
+
+---
+
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: wso2apim-with-analytics-tenants-volume-claim
```
Comment (chirangaalwis) [inline helm/pattern-1/apim-with-analytics/templates/wso2apim-volume-claim.yaml]: @mushthaq33 Apart from the Persistent Volume Claim you have introduced here, you need to introduce a Persistent Volume matching the labels defined in the selector and the resource request (storage) of the Claim, at [1].

And you may also have to introduce options to allow the user to edit NFS configurations via the values.yaml. Please see how this done for the shared mount between _<APIM_HOME>/repository/deployment/server_ directory from [2].

[1]: [Persistent Volume definition](https://github.com/wso2/kubernetes-apim/blob/ba885e0e133ca3995789a7d3745f06cdfa894753/helm/pattern-1/apim-with-analytics/templates/persistent-volumes.yaml)
[2]: [values.yaml](https://github.com/wso2/kubernetes-apim/blob/ba885e0e133ca3995789a7d3745f06cdfa894753/helm/pattern-1/apim-with-analytics/values.yaml#L7-L8)

Comment (mushthaq33) [reply helm/pattern-1/apim-with-analytics/templates/wso2apim-volume-claim.yaml]: Fixed with e343483536bc9a81e58e186eb1e278dec401ee7a

File: helm/pattern-1/apim-with-analytics/templates/persistent-volumes.yaml [hunk 2]
```diff
@@ -28,3 +28,22 @@ spec:
...
+    purpose: apim-shared-tenants
+spec:
+  capacity:
+    storage: 1Gi
+  accessModes:
+    - ReadWriteMany
+  persistentVolumeReclaimPolicy: Retain
+  nfs:
+    server: {{ .Values.serverIp }}
+    path: {{ .Values.sharedTenantsLocationPath }}
```
Comment (chirangaalwis) [inline helm/pattern-1/apim-with-analytics/templates/persistent-volumes.yaml]: @mushthaq33 Please add a new line.

Comment (mushthaq33) [reply helm/pattern-1/apim-with-analytics/templates/persistent-volumes.yaml]: Fixed with b0c46cbe871c139d607bab31ebab3b4cc24e163a

**Pull request merged**